include/alpaka/core/Common.hpp

```diff
@@ -127,3 +127,28 @@
 #else
     #define ALPAKA_STATIC_ACC_MEM_CONSTANT
 #endif
+
+
+// ALPAKA_LAMBDA defines compiler dependent host-device annotation and capture
+// - from: https://github.com/kokkos/kokkos/wiki/Lambda-Dispatch
+#if BOOST_COMP_NVCC || BOOST_COMP_CLANG_CUDA
+
+#if BOOST_COMP_NVCC < BOOST_VERSION_NUMBER(8, 0, 0)
+  #define ALPAKA_FN_LAMBDA [=] __device__
+  #define ALPAKA_FN_LAMBDA_CAPTURE(...) [=,__VA_ARGS__] __device__
+#else
+  #define ALPAKA_FN_LAMBDA [=] __host__ __device__
+  #define ALPAKA_FN_LAMBDA_CAPTURE(...) [=,__VA_ARGS__] __host__ __device__
+  #if ALPAKA_CXX_STANDARD>=17
```

Comment:
`ALPAKA_CXX_STANDARD` is (and probably should not be) added as a preprocessor definition.
You have to rely on a different mechanism to find out if this is allowed.

Reply:
no more using it. Removed class-specific C++17 capture, too. Not required at the moment.